ports/psoc6/Makefile

```diff
@@ -81,77 +95,63 @@ CXXFLAGS += $(CXXFLAGS_MOD)
 
 # Flags for user C modules
 CFLAGS  += $(CFLAGS_MOD)
-CFLAGS  += -DFFCONF_H=\"lib/oofatfs/ffconf.h\" -DLFS1_NO_MALLOC -DLFS1_NO_DEBUG -DLFS1_NO_WARN -DLFS1_NO_ERROR -DLFS1_NO_ASSERT -DLFS2_NO_MALLOC -DLFS2_NO_DEBUG -DLFS2_NO_WARN -DLFS2_NO_ERROR -DLFS2_NO_ASSERT
+
+# Flags from extmod
+CFLAGS  += $(CFLAGS_THIRDPARTY) $(CFLAGS_EXTMOD)
+
 
 LDFLAGS += $(LDFLAGS_MOD)
 
 LIBS =
 
 
-MPY_SRC_C += lib/littlefs/lfs1.c lib/littlefs/lfs1_util.c lib/littlefs/lfs2.c lib/littlefs/lfs2_util.c \
-		 	 lib/oofatfs/ff.c lib/oofatfs/ffunicode.c \
-		 	 \
-		 	 shared/readline/readline.c \
-			 \
-			 shared/runtime/gchelper_native.c \
-			 shared/runtime/interrupt_char.c \
-		 	 shared/runtime/pyexec.c \
-		 	 shared/runtime/stdout_helpers.c \
-			 \
-		 	 shared/timeutils/timeutils.c \
-		     \
-			 shared/runtime/sys_stdio_mphal.c 
-
+MPY_SRC_C = shared/readline/readline.c \
+			\
+			shared/runtime/gchelper_native.c \
+			shared/runtime/interrupt_char.c \
+		 	shared/runtime/pyexec.c \
+		 	shared/runtime/stdout_helpers.c \
+			\
+		 	shared/timeutils/timeutils.c \
+		    \
+			shared/runtime/sys_stdio_mphal.c \
+			\
+			help.c \
+			\
+    		drivers/machine/psoc6_gpio.c \
+    		drivers/machine/psoc6_i2c.c \
+    		drivers/machine/psoc6_system.c  \
+			\
+			modules/gc/modgc.c \
+			\
+			modules/machine/modmachine.c \
+			modules/machine/machine_i2c.c \
+			modules/machine/machine_pin.c \
+			modules/machine/machine_rtc.c \
+			modules/machine/pins.c \
+			\
+			modules/psoc6/modpsoc6.c \
+			modules/psoc6/psoc6_fatfs.c \
+			modules/psoc6/psoc6_flash.c \
+			modules/psoc6/psoc6_qspi_flash.c \
+			\
+			modules/time/modutime.c \
+			\
+			$(SRC_THIRDPARTY_C) \
+			$(SRC_USERMOD) \
+			$(SRC_EXTMOD_C)
+		 	 
 
 SRC_ASM = shared/runtime/gchelper_thumb1.s
 
 
-SRC_C = \
- 	main.c \
-	\
-	help.c \
-	mphalport.c \
-	\
-    drivers/machine/psoc6_gpio.c \
-    drivers/machine/psoc6_i2c.c \
-    drivers/machine/psoc6_system.c  \
-	\
-	modules/gc/modgc.c \
-	\
-	modules/machine/modmachine.c \
-	modules/machine/machine_i2c.c \
-	modules/machine/machine_pin.c \
-	modules/machine/pins.c \
-	modules/machine/machine_rtc.c \
-	\
-	modules/psoc6/modpsoc6.c \
-	modules/psoc6/psoc6_fatfs.c \
-	modules/psoc6/psoc6_flash.c \
-	modules/psoc6/psoc6_qspi_flash.c \
-	\
-	modules/time/modutime.c \
-	\
-	$(MPY_SRC_C) \
-    \
-	$(BUILD)/frozen_content.c
-
-
-SRC_C    += $(SRC_USERMOD)
+SRC_C    += main.c mphalport.c $(MPY_SRC_C) $(BUILD)/frozen_content.c
 SRC_CXX   = 
 
 
-SRC_QSTR += $(MPY_SRC_C) $(SRC_USERMOD) $(SRC_EXTMOD_C)
-SRC_QSTR += modules/machine/modmachine.c \
-            modules/machine/machine_i2c.c \
-            modules/machine/machine_pin.c \
-			modules/machine/machine_rtc.c \
+SRC_QSTR += $(MPY_SRC_C) \
```

Comment:
Is there any reason to include non micropython modules like our drivers folder or SRC_THIRD_PARTY in the SRC_QSTR ? As I understand only those files whose are including MPY string interning are relevant to parse here. 
The extmod.mk is also taking care of adding all those modules to SRC_QRST sources -> https://github.com/jaenrig-ifx/micropython/blob/88e1256d43db3c78e803cae960dd7577564babaf/extmod/extmod.mk#L61

I would take as a reference the stm32 makefile where they only add to the SRC_QRST all port mpy modules and glue mpy code to hal and drivers, etc,

Reply:
The point is that it does not hurt to check all files for  QSTR content, otherwise something may go missing or, when adding more modules, we always have to double check.  In particular for files included from mpconfigport.h are easily overlooked. But if we had a strict, eg directory related policy as to what must be where if it does / does not contain QST stuff, this could be changed. As of now, we do not have such rule. And the makefile is unnecessarily complicated.

Reply:
SRC_EXTMOD_C is now excluded from MPY_SRC_C, the rest we should discuss and make sure directory specific rules

Reply:
I like somehow the organization there is in the stm32 port for the sources. But definitively, that needs to attend some criteria. 
For example, the drivers folder for me has no dependency with the micropython domain, so keeping this away from the QSTR will make clear conceptual cut. 
Not the case with the modules folder. 
And definitively anything which helps to understand which flags are added because of which sources would be a reason for me to have different SRC variables.

Reply:
Let us discuss and fix the criteria and than make the appropriate changes in the makefile. 
For instance, the pins.h/.c should be in the drivers directory, not under modules, as it contains mostly MTB related stuff. 

Also, the i2c header has a dependency to MPY which is not clean. all under drivers should be plain MTB, port or C but not MPY dependent.  If not done cleanly I question the purpose of this separation altogether.